types/history/index.d.ts

```diff
@@ -19,6 +19,8 @@ export interface History {
     goBack(): void;
     goForward(): void;
     block(prompt?: boolean): UnregisterCallback;
+    block(prompt?: string): UnregisterCallback;
```

Comment:
Better use union types: `block(prompt?: boolean|string|TransitionPromptHook)`

Why? See here last section:
https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html

Reply:
Done